subprojects/core-api/src/main/java/org/gradle/work/InputChanges.java

```diff
@@ -98,5 +99,18 @@
      *
      * @param parameterValue The value of the parameter to query.
      */
-    Iterable<FileChange> getFileChanges(Object parameterValue);
+    Iterable<FileChange> getFileChanges(FileCollection parameterValue);
```

Comment:
To me it would make more sense if we renamed `parameterValue` to `parameter`. I can't fully argue for this, but it's a bit like the `FileCollection` is parameter, and `getFiles()` is its actual value or something. It works a little bit more smoothly with the `Provider` overload. In any case, I don't think `Value` adds anything.

Reply:
Renamed to `parameter`.